### ISSUE TITLE
What should I do if I wanna to add ACRA and public my apps to F-Droid?

### ISSUE THREAD

**Issue by TigerBeanst** [ACRA/acra#1002]
I'm just getting to know ACRA, but I have some questions.

If I want my apps public to F-Droid, and still have crash report and analysis, I would have to include data such as server address and keys in the source code, but this data should theoretically not be public (just like when using Firebase you should not push the Google-services.json file).

What should I do?

**Comment by F43nd1r** [comment]
Anything you package inside an apk should be considered public knowledge as it can be quite easily discovered. You can load the parameters from a file similar to Google-services.json to obscure it a bit, but in general this is just accepted and the server has to sanitize received requests.

**Comment by TigerBeanst** [comment]
> Anything you package inside an apk should be considered public knowledge as it can be quite easily discovered. You can load the parameters from a file similar to Google-services.json to obscure it a bit, but in general this is just accepted and the server has to sanitize received requests.

That's where I get confused. If I include these keys in the source code, it can be found by anyone. But if I don't include it, analysis of my apps won't work.

I've read that some apps released on F-Droid has analysis, are they exposing keys to the public?

**Comment by F43nd1r** [comment]
Yes. Always consider your client applications compromised and do anything critical on server side, that's just how software works.
See also https://guides.codepath.com/android/Storing-Secret-Keys-in-Android#storing-fixed-keys
